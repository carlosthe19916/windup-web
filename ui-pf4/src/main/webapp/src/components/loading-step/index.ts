export { LoadingStep } from "./loading-step";
