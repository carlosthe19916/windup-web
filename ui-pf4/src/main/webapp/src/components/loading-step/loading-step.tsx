import React from "react";
import {
  EmptyState,
  EmptyStateIcon,
  Title,
  Button,
  EmptyStateVariant,
  EmptyStateSecondaryActions,
  Bullseye,
  Spinner,
} from "@patternfly/react-core";

export interface LoadingStepProps {
  customText: string;
  cancelTitle?: string;
  onClose?: () => void;
}

export const LoadingStep: React.FC<LoadingStepProps> = ({
  customText,
  cancelTitle,
  onClose,
}) => {
  return (
    <Bullseye>
      <EmptyState variant={EmptyStateVariant.full} className="pf-u-mt-4xl">
        <EmptyStateIcon icon={Spinner} className="pf-u-mb-0" />
        <Title headingLevel="h2" size="xl" className="pf-u-mt-xl">
          {customText}
        </Title>
        {onClose && (
          <EmptyStateSecondaryActions className="pf-u-mt-xl">
            <Button variant="link" onClick={onClose}>
              {cancelTitle}
            </Button>
          </EmptyStateSecondaryActions>
        )}
      </EmptyState>
    </Bullseye>
  );
};
