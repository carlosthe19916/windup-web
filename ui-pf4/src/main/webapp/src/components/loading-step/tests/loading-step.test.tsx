import React from "react";
import { shallow } from "enzyme";
import { LoadingStep } from "../loading-step";

describe("Welcome", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<LoadingStep />);
    expect(wrapper).toMatchSnapshot();
  });
});
