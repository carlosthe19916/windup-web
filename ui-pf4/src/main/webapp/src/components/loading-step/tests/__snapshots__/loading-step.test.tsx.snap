// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Renders without crashing 1`] = `
<Bullseye>
  <EmptyState
    className="pf-u-mt-4xl"
    variant="full"
  >
    <EmptyStateIcon
      className="pf-u-mb-0"
      icon={[Function]}
    />
    <Title
      className="pf-u-mt-xl"
      headingLevel="h2"
      size="xl"
    />
  </EmptyState>
</Bullseye>
`;
